app-emulation/aranym/aranym-1.0.2.recipe

```diff
@@ -0,0 +1,59 @@
+SUMMARY="Atari Running on Any Machine"
+DESCRIPTION="
+ARAnyM (Atari Running on Any Machine) is a multiplatform virtual machine for \
+running Atari ST/TT/Falcon operating systems and applications.
+"
+HOMEPAGE="http://github.com/aranym/aranym"
+COPYRIGHT="2001-2018 ARAnyM developer team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/aranym/aranym/archive/ARANYM_${portVersion//./_}.tar.gz"
+CHECKSUM_SHA256="c0935aee14cca55c08e75a94b77dcd69f326ae563ae371b22c7b9d8efd82e24e"
+SOURCE_DIR="aranym-ARANYM_${portVersion//./_}"
+PATCHES="aranym-$portVersion.patch"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	aranym = $portVersion
+	app:aranym = $portVersion
+	cmd:aranym = $portVersion
+	"
+REQUIRES="
+	haiku
+	lib:libsdl
+	lib:libSDL_image_1.2
+	lib:libjpeg
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libsdl_1.2
+	devel:libSDL_image_1.2
+	devel:libjpeg
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
```

Comment:
```diff
-	cmd:aclocal
-	cmd:autoconf
+	cmd:autoreconf

Reply:
autoreconf does not work, because the project does not use automake, and that is the only tool that will install those files.

Reply:
Shouldn't --libtoolize -fci trigger this? (if not done by autoreconf)

Reply:
libtool isn't use either...

Reply:
`libtoolize` won't panic over a program that doesn't use `libtool`, so it's somewhat better than ignoring `automake` exit code.

New suggestions:

```diff
 	cmd:autoconf
+	cmd:autoheader
+	cmd:libtoolize
```

`autoheader` was added due to being used by `./autogen.sh`

Reply:
>libtoolize won't panic over a program that doesn't use libtool, so it's somewhat better than >ignoring automake exit code.

But that will pull in dependencies and files that aren't really used by the project. Also, the libtool package has it own versions of config.{guess,sub}. I can't see why that should be better than the previous approach. In the current version (from git), automake will be used anyway.

> autoheader was added due to being used by ./autogen.sh

But isn't that part of the same package (autoconf)? Maybe better to add an dependency to the package instead?

Reply:
Although additng a dependency on the autoconf package would work perfectly, it is better to add dependencies on each of the `cmd:` used in the recipe.
This could be useful in case `cmd:autoheader` gets moved to a subpackage or to another package.
Here I guess `./autogen.sh` uses both autoconf and autoheader, so we need at least `cmd:autoheader`.
And, if I'm not mistaken, autoheader does not need autoconf, so we also need to mention `cmd:autoconf`, otherwise autoconf would no longer be available in the chroot if `cmd:autoheader` gets moved to a subpackage which does not pull `cmd:autoconf`.

Reply:
@th-otto Then I believe it's fine to use `automake`, but please add `autoheader` to build reqs.

Reply:
@fbrosson would be strange to split autoconf/autoheader since they need the same Autom4te machinery ;) Nevertheless, added the requested changes.

A bit offtopic: i also tried to get the JIT compiler working. That configuration (due to design) needs a segmentation violation handler that emulates a few instructions of the host. Generally, the porting was easy, just had to take into account the different layout of the ucontext_t structure. However, when i try to run this, it panics the kernel... any idea how that is possible? It i just user-code that is executed.

A simple test-program that tries to mimic the behaviour works, though.

Reply:
I'm affraid I can't help you with these questions :(
Hopefully there are several other contributors who will certainly be able to answer your questions :)

Reply:
Please open a bug report at https://dev.haiku-os.org/ and attach the KDL output there. Userland programs should indeed not be able to crash the kernel, so it is most likely a bug.